timelock-server/src/integTest/java/com/palantir/atlasdb/timelock/AsyncTimelockServiceIntegrationTest.java

```diff
@@ -0,0 +1,155 @@
+/*
+ * Copyright 2017 Palantir Technologies, Inc. All rights reserved.
+ *
+ * Licensed under the BSD-3 License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://opensource.org/licenses/BSD-3-Clause
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.timelock;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.junit.Assert.assertTrue;
+
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.Future;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+import org.junit.ClassRule;
+import org.junit.Test;
+import org.junit.rules.RuleChain;
+
+import com.google.common.collect.ImmutableSet;
+import com.palantir.lock.LockDescriptor;
+import com.palantir.lock.StringLockDescriptor;
+import com.palantir.lock.v2.LockImmutableTimestampRequest;
+import com.palantir.lock.v2.LockImmutableTimestampResponse;
+import com.palantir.lock.v2.LockRequestV2;
+import com.palantir.lock.v2.LockTokenV2;
+import com.palantir.lock.v2.WaitForLocksRequest;
+
+public class AsyncTimelockServiceIntegrationTest {
+    private static final String CLIENT = "test";
+
+    private static final TestableTimelockCluster CLUSTER = new TestableTimelockCluster(
+            "http://localhost",
+            CLIENT,
+            "paxosSingleServer.yml");
+
+    @ClassRule
+    public static final RuleChain ruleChain = CLUSTER.getRuleChain();
+
+    private static final LockDescriptor LOCK_A = StringLockDescriptor.of("a");
+    private static final LockDescriptor LOCK_B = StringLockDescriptor.of("b");
+
+    private final ExecutorService executor = Executors.newCachedThreadPool();
+
+    @Test
+    public void canLockRefreshAndUnlock() {
+        LockTokenV2 token = CLUSTER.lock(requestFor(LOCK_A));
+        boolean wasRefreshed = CLUSTER.refreshLockLease(token);
+        boolean wasUnlocked = CLUSTER.unlock(token);
+
+        assertTrue(wasRefreshed);
+        assertTrue(wasUnlocked);
+    }
+
+    @Test
+    public void locksAreExclusive() {
+        LockTokenV2 token = CLUSTER.lock(requestFor(LOCK_A));
+        Future<LockTokenV2> futureToken = executor.submit(() -> CLUSTER.lock(requestFor(LOCK_A)));
```

Comment:
yep, could have that.. the endpoint is actually defined as a blocking API, but yeah i can see how that might make testing easier to move the executor into the cluster class